ipatests/test_integration/test_commands.py

```diff
@@ -783,3 +859,65 @@ def test_sss_ssh_authorizedkeys(self):
             backup.restore()
             self.master.run_command(['rm', '-f', pem_file, user_key,
                                      '{}.pub'.format(user_key)])
+
+    def test_cacert_manage(self):
+        """Exercise ipa-cacert-manage delete"""
+
+        # deletion without nickname
+        result = self.master.run_command(
+            ['ipa-cacert-manage', 'delete'],
+            raiseonerr=False
+        )
+        assert result.returncode != 0
+
+        # deletion with an unknown nickname
+        result = self.master.run_command(
+            ['ipa-cacert-manage', 'delete', 'unknown'],
+            raiseonerr=False
+        )
+        assert result.returncode != 0
+        assert "Unknown CA 'unknown'" in result.stderr_text
+
+        # deletion of IPA CA
+        ipa_ca_nickname = get_ca_nickname(self.master.domain.realm)
+        result = self.master.run_command(
+            ['ipa-cacert-manage', 'delete', ipa_ca_nickname],
+            raiseonerr=False
+        )
+        assert result.returncode != 0
+        assert 'The IPA CA cannot be removed with this tool' in \
+               result.stderr_text
+
+        # Install 3rd party CA's, Let's Encrypt in this case
+        for cert in (isrgrootx1, letsencryptauthorityx3):
+            with tempfile.NamedTemporaryFile() as f:
+                f.write(cert)
```

Comment:
This command is run on the controller, not on the master. As a result, the file exists only on the controller and ipa-cacert-manage does not find the file. The method self.master.put_file_contents() can be used instead.